### PR TITLE
#18069 : Updating /loginas and /logoutas methods to correctly revert …

### PR DESCRIPTION
**Pull request by jcastro-dotcms**
…Session data changes when a given user could not be impersonated. Data was not being re-set correctly as the original user, which was causing errors in the back-end that required the user to log out and log back in. Removing unnecessary code. Updating Javadoc.

### CODE REVIEW

**Review by freddyucv — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/user/UserResource.java [hunk 1]
```diff
@@ -495,33 +521,32 @@ public final Response logoutAs(@Context final HttpServletRequest httpServletRequ
...
 					.init();
 
-			User currentUser = initData.getUser();
-
+			currentUser = initData.getUser();
 			response = Response.ok(this.helper.getLoginAsUsers(currentUser, filter, includeUsersCount)).build();
-		} catch (Exception e) {
-			SecurityLogger.logInfo(UserResource.class, "An error occurred when processing the request. " + e.getMessage());
-			response = ExceptionMapperUtil.createResponse(e, Response.Status.INTERNAL_SERVER_ERROR);
+		} catch (final Exception e) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/user/UserResource.java]: Codacy Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5162183)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/user/UserResource.java [hunk 2]
```diff
@@ -342,68 +340,104 @@ public final Response loginAs(@Context final HttpServletRequest request, @Contex
...
+	 */
+	private void updateLoginAsSessionInfo(final HttpServletRequest request, final Host site, final String
+			principalUserId, final String loginAsUserId) {
+		final HttpSession session = request.getSession();
+		if (UtilMethods.isSet(principalUserId)) {
+			session.setAttribute(WebKeys.PRINCIPAL_USER_ID, principalUserId);
+		} else {
+			session.setAttribute(WebKeys.PRINCIPAL_USER_ID, null);
+		}
+		final String userToImpersonate = (UtilMethods.isSet(loginAsUserId) ? loginAsUserId : principalUserId);
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/user/UserResource.java]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5162183)

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/user/UserResource.java [hunk 3]
```diff
@@ -342,68 +339,119 @@ public final Response loginAs(@Context final HttpServletRequest request, @Contex
...
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/user/UserResource.java]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5162183)

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/user/UserResource.java [hunk 4]
```diff
@@ -495,33 +535,32 @@ public final Response logoutAs(@Context final HttpServletRequest httpServletRequ
...
 					.init();
 
-			User currentUser = initData.getUser();
-
+			currentUser = initData.getUser();
 			response = Response.ok(this.helper.getLoginAsUsers(currentUser, filter, includeUsersCount)).build();
-		} catch (Exception e) {
-			SecurityLogger.logInfo(UserResource.class, "An error occurred when processing the request. " + e.getMessage());
-			response = ExceptionMapperUtil.createResponse(e, Response.Status.INTERNAL_SERVER_ERROR);
+		} catch (final Exception e) {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/user/UserResource.java]: Codacy Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5162183)

**Pull request merged**